packages/toolkit/src/lib/react-query-service/model/use-namespace-models/server.ts

```diff
@@ -0,0 +1,69 @@
+import type { Nullable, Visibility } from "instill-sdk";
+
+import { env, QueryClient } from "../../../../server";
+import { getInstillModelAPIClient } from "../../../vdp-sdk";
+import { queryKeyStore } from "../../queryKeyStore";
+
+export async function fetchNamespaceModels({
+  namespaceId,
+  accessToken,
+  filter,
+  visibility,
+}: {
+  namespaceId: Nullable<string>;
+  accessToken: Nullable<string>;
+  filter: Nullable<string>;
+  visibility: Nullable<Visibility>;
+}) {
+  try {
+    if (!namespaceId) {
+      throw new Error("namespaceId not provided");
+    }
+
+    const client = getInstillModelAPIClient({
+      accessToken: accessToken ?? undefined,
+    });
+
+    const models = await client.model.listNamespaceModels({
+      namespaceId,
+      filter: filter ?? undefined,
+      visibility: visibility ?? undefined,
+      pageSize: env("NEXT_PUBLIC_QUERY_PAGE_SIZE"),
+      enablePagination: false,
+    });
+
+    return Promise.resolve(models);
+  } catch (err) {
+    return Promise.reject(err);
+  }
+}
+
+export function prefetchNamespaceModels({
```

Comment:
When is it used?

Reply:
Currently no, but if we really want to do so SSR prefetch stuff, this will serves the need